src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -133,13 +133,13 @@ void BattlegroundMgr::Update(uint32 diff)
         std::vector<uint64> scheduled;
         std::swap(scheduled, m_QueueUpdateScheduler);
 
-        for (uint8 i = 0; i < scheduled.size(); i++)
+        for (unsigned long i : scheduled)
         {
-            uint32 arenaMMRating = scheduled[i] >> 32;
-            uint8 arenaType = scheduled[i] >> 24 & 255;
-            BattlegroundQueueTypeId bgQueueTypeId = BattlegroundQueueTypeId(scheduled[i] >> 16 & 255);
-            BattlegroundTypeId bgTypeId = BattlegroundTypeId((scheduled[i] >> 8) & 255);
-            BattlegroundBracketId bracket_id = BattlegroundBracketId(scheduled[i] & 255);
+            uint32 arenaMMRating = i >> 32;
```

Comment:
```cpp
C:\projects\trinitycore\src\server\game\Battlegrounds\BattlegroundMgr.cpp(138): error C2220: warning treated as error - no 'object' file generated [C:\projects\trinitycore\build\src\server\game\game.vcxproj]
C:\projects\trinitycore\src\server\game\Battlegrounds\BattlegroundMgr.cpp(138): warning C4293: '>>': shift count negative or too big, undefined behavior [C:\projects\trinitycore\build\src\server\game\game.vcxproj]
```

Reply:
Can you share your cmake params so I can steal and use the same?

This is very weird code, I'll look into it

Reply:
This is based on the CMake parameters used in AppVeyor:
https://ci.appveyor.com/project/DDuarte/trinitycore/builds/30668901#L9

Reply:
This breaks because clang and msvc disagree on size of unsigned long.
For clang `unsigned long` is 64 bits (uint64)
For msvc `unsigned long` is 32 bits (uint32)

Clearly not using our uint64 typedef (which is guaranteed 64 bits on all platforms) breaks it

Reply:
Also notice that AppVeyor is set to make & build a 32 bits core.

Reply:
No, it is not. Don't spread misinformation

Reply:
OK, well, I see. so `DCMAKE_C_FLAGS="/DWIN32` is irrelevant, I guess?

Reply:
It is. That define is a generic compile definition that various third party libraries use to detect that they are compiled for windows, not specifically 32 bit windows

Reply:
I'm still willing to go through the diff and correct these where they're wrong, along with the autos. If you want the change in TC once it's done (probably will make other PR's per folder or smth to get less bulk work, about 150 changes each or smth).